### PR TITLE
Adding DSC template and resource Submodules

### PR DESCRIPTION
**Pull request by gaelcolas**
# Pull Request

## Pull Request (PR) description

Adding `DSC_Folder` resource as per former `DscResource.Template`.
Leaving only Get/Set/Test in `DSC_Folder.psm1`, the rest goes in `Folder.Common.psm1` module.
Using the `DscResource.Common` module from PS Gallery as (reusable) artefact.

leaving this as Draft as I now need to add a task to `Invoke-DscResourceTest` to validate the resource is HQRM

### CODE REVIEW

File: tests/Unit/DSCResources/DSC_Folder.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,553 @@
+#region HEADER
+$script:dscModuleName = 'Sampler'
+$script:dscResourceName = 'DSC_Folder'
+
+$script:moduleRoot = Split-Path -Parent (Split-Path -Parent $PSScriptRoot)
+
+Import-Module -Name DscResource.Test -Force
+
+$TestEnvironment = Initialize-TestEnvironment `
```
Comment (johlju) [inline tests/Unit/DSCResources/DSC_Folder.Tests.ps1]: Maybe we can move this into `Invoke-TestSetup`?

**Comment by johlju** [comment]
This looks good to me. 

I did just browse the code, but as I browsed I thought that there is a lot of code in the tasks. Maybe we could split the code in the tasks into units (functions) so we can test the code easier and would make the tasks more readable maybe. 🤔  This can be done in a future PR. 🙂

**Comment by gaelcolas** [comment]
Agreed that the code should be split, but I think that will have to be in "Yet another module" to keep Sampler "simplistic".
Yeah, it's a bit quick and dirty.

**Comment by johlju** [comment]
@gaelcolas Does not seem that the DSCResource examples are copied when running Plaster? That file does not seem to be updated? 🤔

**Comment by johlju** [comment]
Seems it are commented

https://github.com/gaelcolas/Sampler/blob/7f226a81eb307de75e034b678c5a81944a0ca686/Sampler/Templates/Sampler/plasterManifest.xml#L296-L318

**Comment by johlju** [comment]
The asset for RequiredModules.ps1 needs to be updated with `'DscResource.Test'   = 'latest'`?

**Comment by gaelcolas** [comment]
Well spotted. On it

**Comment by gaelcolas** [comment]
Fingers crossed, but that should be ok...

File: Sampler/Assets/build.yaml [hunk 2]
```diff
@@ -49,6 +50,7 @@ BuildWorkflow:
   test:
```
Comment (johlju) [inline Sampler/Assets/build.yaml]: We should add `    - build` to the `pack` task.

Comment (gaelcolas) [reply Sampler/Assets/build.yaml]: done

**Pull request merged**